dask/array/tests/test_slicing.py

```diff
@@ -616,6 +625,23 @@ def crash():
         result.compute()
 
 
+def test_flat_chunks():
```

Comment:
We don't usually add tests for non-public simple helper functions, as they complicate changing dask's internals later on (if I delete/change this method, I also have to delete/change this test). Testing the user-facing api in a way that covers the cases you care about should be enough here.

Reply:
So I almost didn't add a test for this private helper function, but in the end (a) it seemed like good due diligence (as in, it's what we'd do in a SciTools repo, and they're the rules I'm most used to!), (b) I needed to check the function was doing what I thought it was, and (c) the behaviour this adds seems quite hard to isolate given how this function is called.

In any case, with the updated logic this private helper function is no longer needed, so I've removed the function and its test...